### ISSUE TITLE
Micro average for auroc

### ISSUE THREAD

**Issue by FlorianMF** [PyTorchLightning/metrics#104]
## 🚀 Feature
Add 'micro' to list of allowed averages.

### Motivation
For now, 'average' must be in [None, 'macro', 'weighted].
For multi-label classification it would be nice to allow micro averaging. 
This comes down to calculating `auroc(preds.flatten(), target.flatten())`.

Similarly to https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html.
One could also consider adding 'samples' to the list of accepted averages.

What do you think?
